### PR TITLE
Fix a bug that required is not inherited by Parent DSL

### PR DESCRIPTION
**Pull request by ikawaha**
see. #2429

### CODE REVIEW

File: expr/http_endpoint.go [hunk 1]
```diff
@@ -192,18 +192,23 @@ func (e *HTTPEndpointExpr) Prepare() {
...
 						}
-						if o := AsObject(e.MethodExpr.Payload.Type); o != nil {
-							if o.Attribute(name) == nil {
-								o.Set(name, att)
+						if o := AsObject(e.MethodExpr.Payload.Type); o != nil && o.Attribute(name) == nil {
+							if c.MethodExpr.Payload.IsRequired(name) {
+								if e.MethodExpr.Payload.Validation == nil {
+									e.MethodExpr.Payload.Validation = &ValidationExpr{}
+								}
+								e.MethodExpr.Payload.Validation.Required = append(e.MethodExpr.Payload.Validation.Required, name)
```
Comment (raphael) [inline expr/http_endpoint.go]: There is a method that makes this easier, it takes care of removing duplicate entries in the `Required` field: https://github.com/goadesign/goa/blob/21aa53219da0de90202ca13f99c612bcb594c817/expr/attribute.go#L630

**Review by raphael — COMMENTED** [review]
This is great! thank you very much! I left a comment for a small change.

**Comment by ikawaha** [comment]
Thank you for your suggestion!

**Comment by raphael** [comment]
Looks great, thank you!

**Pull request merged**